jenkins/opensearch/maven-publish-1.3.x.jenkinsfile

```diff
@@ -0,0 +1,90 @@
+lib = library(identifier: 'jenkins@1.5.4', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/opensearch-project/opensearch-build-libraries.git',
+]))
+
+pipeline {
+    options {
+        timeout(time: 2, unit: 'HOURS')
+    }
+    agent none
+    environment {
+        AGENT_X64 = 'Jenkins-Agent-AL2-X64-C54xlarge-Docker-Host'
+    }
+    triggers {
+        parameterizedCron '''
+            H/60 * * * * %INPUT_MANIFEST=1.3.10/opensearch-1.3.10.yml
+        '''
+    }
+    parameters {
+        string(
+            name: 'COMPONENT_NAME',
+            description: 'If this field contains one or more component names (e.g. OpenSearch common-utils ...), will build with "--component <COMPONENT_NAME> ...", else build everything in the INPUT_MANIFEST.',
+            trim: true
+        )
+        string(
+            name: 'INPUT_MANIFEST',
+            description: 'Input manifest under the manifests folder, e.g. 2.0.0/opensearch-2.0.0.yml.',
+            trim: true
+        )
+    }
+    stages {
+        stage('dockerAgent-setup') {
+            agent {
+                docker {
+                    label AGENT_X64
+                    image 'docker/library/alpine:3'
+                    registryUrl 'https://public.ecr.aws/'
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    echo("Detect Docker Images and Related Parameters")
+                    dockerAgent = detectDockerAgent()
+                    currentBuild.description = INPUT_MANIFEST
+                }
+            }
+        }
+        stage('publish-maven-snaphots') {
+            environment {
+                SNAPSHOT_REPO_URL = "https://aws.oss.sonatype.org/content/repositories/snapshots/"
+            }
+            agent {
+                docker {
+                    label AGENT_X64
+                    image dockerAgent.image
+                    args dockerAgent.args
+                    alwaysPull true
+                }
+            }
+            steps {
+                script {
+                    buildManifest(
+                        componentName: "${COMPONENT_NAME}",
+                        inputManifest: "manifests/${INPUT_MANIFEST}",
+                        platform: 'linux',
+                        architecture: 'x64',
+                        distribution: 'tar',
+                        snapshot: true
+                    )
+
+                    String mavenPath = "$WORKSPACE/tar/builds/opensearch/maven"
+
+                    if (fileExists(mavenPath)) {
+                        withCredentials([usernamePassword(credentialsId: 'jenkins-sonatype-creds', usernameVariable: 'SONATYPE_USERNAME', passwordVariable: 'SONATYPE_PASSWORD')]) {
```

Comment:
Can we use https://github.com/opensearch-project/common-utils/blob/main/.github/workflows/maven-publish.yml#L34-L35 credentials for snapshots?

Reply:
Updated Sayali